### DISCUSSION TITLE
Please do not ever enable Enterprise Policies 🙏

### DISCUSSION THREAD

**Discussion by rubaboo** [zen-browser/desktop#2195] [Ideas]
Firefox and LibreWolf are both unusable in a corporate environment, due to policies preventing things like installing extensions, having a persistent session ("Open previous windows and tabs"), etc.

I was getting by with Vivaldi for a while, but I can't stand the Chromium family. Been using Firefox since it was Netscape Navigator.

**Comment by pinage404** [comment]
My company had enabled Enterprise Policies

We connect to ours internal tools using Google SSO, i can't use anything else than Google Chrome

I would love to be able to use Zen at work

**Reply by rubaboo** [reply]
I'm hoping devs understand what I mean, but perhaps "enable" was not a clear enough word to use? Yes, companies implement Enterprise Policies, crippling user experience in the name of "safety". It is up to the browser to be confined by those policies or to ignore them.

Firefox and LibreWolf are same in this respect, they are both bound by the policies, not allowing me to install any extensions or even start with the same windows and tabs that were open last time. Vivaldi and Zen, on the other hand, are not paying attention to those policies.

I'd very much like Zen to continue disregarding Enterprise Policies.

**Reply by rubaboo** [reply]
@pinage404 I am not familiar with Google SSO, but if you think Enterprise Policies support would allow you to use a Firefox-based browser at work, then you should try using Firefox or LibreWolf, because they do support Enterprise Policies. Are you able to use Firefox, but want Zen specifically?

**Reply by pinage404** [reply]
At work, i can't use Firefox either, i can log in with Google but once i'm logged in, i get a page that prevents me from accessing the service

**Reply by rubaboo** [reply]
If you can't use Firefox, which already has Enterprise Policies functionality, how is enabling this functionality in Zen, which is a fork of Firefox, going to help you? Do you even understand that "Enterprise Policies" here refers to a particular feature of Firefox? Not to some limitation that your company put in place to force using Chrome. Which could probably be done in a multitude of different ways, many of which Zen could not possibly hope to bypass or comply with.

**Comment by cheeeebs** [comment]
Does your IT team just allow the use of any browser you want?

**Reply by Midrielen** [reply]
I just cracked the BIOS, installed a free system and called my day. I have the Windows installation transferred to a USB stick, so I can restore it in any moment (it would be slow though). If the IT team knows I think they are just relieved, because now I don't share my FOSS ideology with them anymore.

**Comment by ingoau** [comment]
Technically it is enabled, it's just the browser isn't well know enough for any IT admins to implement policies.

Basically, firefox and Firefox based browsers (on windows) get policies from HKLM\Software\Policies\Mozilla\APP NAME. In Firefox, the app name is Firefox so the policies would be in HKLM\Software\Policies\Mozilla\Firefox, and after a search through Zen's codebase, the app name is just Zen so the path is HKLM\Software\Policies\Mozilla\Zen

**Reply by rubaboo** [reply]
Oh shoot. So it's just a matter of time, potentially. I need a switch to turn them off!
Or, is there a way to change the APP NAME? A userChrome javascript perhaps?

**Reply by ingoau** [reply]
> Or, is there a way to change the APP NAME? A userChrome javascript perhaps?

You would have to modify the source and rebuild every update

**Reply by ingoau** [reply]
> So it's just a matter of time

This is still quite a small project (at least compared to Firefox itself) so I don't think this will be implemented by most IT admins, either because they don't have the time, don't care, or don't know about Zen. As an IT admin I found this out when I was searching through Mozilla's and Zen's codebase, but didn't implement because no one else uses Zen and I was really just interested in how to do this.

**Reply by rubaboo** [reply]
> You would have to modify the source and rebuild every update

That will probably trigger Windows Defender (or whatever other "protection", I have more than one installed on work laptop).

**Reply by Mecallie** [reply]
I suggest you talk to your IT department as to why certain policies need to be in place. If it is not about security, maybe you can convince them to drop them. If it is because of security maybe they can convince you to be safe, rather than sorry, when working with company assets and information.

**Comment by miyl** [comment]
Policies can be useful to manage one's own machine(s) - but forcing them on other's seems really invasive.

I wonder if most portable versions of browsers don't read the policies at their standard locations?  
If an application is sufficiently sandboxed maybe it won't even have the ability to read those locations...?

**Reply by ingoau** [reply]
> I wonder if most portable versions of browsers don't read the policies at their standard locations?  

They probably do, unless the source has been modified to not

> If an application is sufficiently sandboxed maybe it won't even have the ability to read those locations...?

Yes, but you would have to install a sandboxing solution which would probably not be allowed

**Reply by rubaboo** [reply]
> Policies can be useful to manage one's own machine(s) - but forcing them on other's seems really invasive.

Ditto. Unfortunately, the only difference between the two is whether one has admin access on the machine or not.

**Comment by juju4** [comment]
I would encourage you, your manager or anyone appropriate to have a constructive discussion with your security/IT team to allow some extensions with allowlist mode present in Firefox. With a good relationship, you can make that a win-win. If already, try again.
Depending on your enterprise context, it can apply easily or not.
If you are in a sensitive sector or Defense, probably not. See recent news...

**Comment by itpropro** [comment]
I can understand that you maybe don’t like the policies your IT team has implemented (which should also apply to Chrome btw), but this is an essential feature for a mature browser.
We all want Zen to grow in adoption and part of that is making it packagable and managable via. enterprise tooling.

**Comment by Mecallie** [comment]
I manage the IT security for a large international company. We do not want to restrict users if not needed, but we do need to protect our infrastructure and prevent users from "doing something stupid". We also need to make sure everything is as secure as possible.

Some of the things I have done:
Make sure only safe protocols are used (no old TLS/SSL connections).
Disable 3rd party cookies.
Import Enterprise Root Certificates
Install an adblocker (Google really f*cked this with the manifest 3 reqs., if someone knows how to get ublock origin to work let me know!)
Allow SPNEGO/NTLM authentication for certain (internal) sites.
Set a default home page (intranet) which is changeable by the user.

Users were very happy with the add blocker, and the fact they now have SSO for a lot more sites so they do not have to log in.
I am not using it to force users to use a certain browser, nor am I using it to force all kinds of preferences unrelated to security for our users.

But we do, for instance, want the option to block specific extensions/add-ons from running, like VPN connections and such. Not having any management over your browser simply means no big company will use it, period.

Fortunately it is quite simple to implement. I have done so for the Floorp browser and now did it for the Zen browser as well.
You can use the attached .admx and .adml files to manage Zen via your enterprise group policies (GPO). Note: this is a pretty basic "search and replace" action. I did test it (add-on install and search engine setting), but there might be (cosmetic) issues.

Feel free to use at your own risk. And at the Zen team: I am happy to maintain these policy files if you provide me with some feedback.
[zen.zip](https://github.com/user-attachments/files/22613572/zen.zip)

**Reply by Mecallie** [reply]
One quick remark: the search does not work as expected. When I enter %s at the end of the search url it only searches for %s when I search from the navigation (new tab) window. When I do not add it it does not search for anything I typed but opens the search engine. Strange behaviour.

**Reply by masterned** [reply]
@Mecallie, according to the Mozilla policies template: https://mozilla.github.io/policy-templates/#searchengines--add , instead of `%s`, you'll want to use `{searchTerms}` for the substitution string.

**Reply by Mecallie** [reply]
> @Mecallie, according to the Mozilla policies template: https://mozilla.github.io/policy-templates/#searchengines--add , instead of `%s`, you'll want to use `{searchTerms}` for the substitution string.

It was indeed this simple! Policies seem to work just fine, happy camper.
